src/main/java/org/reaktivity/nukleus/ws/internal/routable/stream/TargetOutputEstablishedStreamFactory.java

```diff
@@ -269,6 +270,14 @@ private void processEnd(
                     {
                         headers.item(h -> h.name("sec-websocket-protocol").value(negotiated));
                     }
+                    else if(protocol != null)
+                    {
+                        headers.item(h -> h.name("sec-websocket-protocol").value(protocol));
+                    }
```

Comment:
Remove this code block and default `negotiated` to `protocol` when `wsProtocol` is `null`.